Resources/AzureDevOps/JobTemplates/deploy-environment.yml

```diff
@@ -49,122 +49,35 @@ jobs:
-
-- deployment: GetAllSkillsForOccupation
-  pool: 'NCS - CI and CD'
-  displayName: "Deploy GetAllSkillsForOccupation function"
-  environment: DEV_SERVICETAXONOMY
-  dependsOn: 
-  - DeployTemplate
-  - SetupConditions
-  condition: or(eq(dependencies.SetupConditions.outputs['setupConditionStep.projectsToDeploy'], '*'), contains(dependencies.SetupConditions.outputs['setupConditionStep.projectsToDeploy'], 'GetAllSkillsForOccupation'))
-  strategy:
-    runOnce:
-      deploy:
-        steps:
-        - task: DownloadBuildArtifacts@0
-          displayName: "Download build artifacts"
-          inputs:
-            buildType: 'current'
-            artifactName: 'DFC.ServiceTaxonomy.ApiFunction'  
-        - template: ../StepTemplates/deploy-azure-function.yml
-          parameters:
-            FunctionType: 'GetAllSkillsForOccupation'
-            WebAppName: '${{ parameters.WebAppPrefix }}-stax-skillsocc-fa'
-            AzureSubscription: '${{ parameters.AzureSubscription }}'
-            SharedResourceGroup: '${{ parameters.SharedResourceGroup }}'
-            ApimServiceName: '${{ parameters.ApimServiceName }}'
-            ApimImportScript: '${{ parameters.ApimImportScript }}'
-            SpecificationPath:  '${{ parameters.SpecificationPath }}'
-            FunctionAppDomain: '${{ parameters.FunctionAppDomain }}'
-            PackageFile: '${{ parameters.PackageFile }}'
-            ResourceGroup: '${{ parameters.ResourceGroup }}'
-
-- deployment: GetOccupationsByLabel
-  pool: 'NCS - CI and CD'
-  displayName: "Deploy GetOccupationsByLabel function"
-  environment: DEV_SERVICETAXONOMY
-  dependsOn:
-  - DeployTemplate
-  - SetupConditions
-  condition: or(eq(dependencies.SetupConditions.outputs['setupConditionStep.projectsToDeploy'], '*'), contains(dependencies.SetupConditions.outputs['setupConditionStep.projectsToDeploy'], 'GetOccupationsByLabel'))
-  strategy:
-    runOnce:
-      deploy:
-        steps:
-        - task: DownloadBuildArtifacts@0
-          displayName: "Download build artifacts"
-          inputs:
-            buildType: 'current'
-            artifactName: 'DFC.ServiceTaxonomy.ApiFunction'  
-        - template: ../StepTemplates/deploy-azure-function.yml
-          parameters:
-            FunctionType: 'GetOccupationsByLabel'
-            WebAppName: '${{ parameters.WebAppPrefix }}-stax-occbylabel-fa'
-            AzureSubscription: '${{ parameters.AzureSubscription }}'
-            SharedResourceGroup: '${{ parameters.SharedResourceGroup }}'
-            ApimServiceName: '${{ parameters.ApimServiceName }}'
-            ApimImportScript: '${{ parameters.ApimImportScript }}'
-            SpecificationPath:  '${{ parameters.SpecificationPath }}'
-            FunctionAppDomain: '${{ parameters.FunctionAppDomain }}'
-            PackageFile: '${{ parameters.PackageFile }}'
-            ResourceGroup: '${{ parameters.ResourceGroup }}'
\ No newline at end of file
+- ${{ each apiPartition in parameters.Apis }}:
```

Comment:
We introduced a new parameter called 'Apis' above,  which is an array of arrays of objects.

Each outer array represents a partitionable block of deploys,  and the inner array is the jobs inside that partition. 

This block iterates over that,  adding a deployment job for each API.